subsys/bluetooth/mesh/Kconfig

```diff
@@ -93,68 +93,95 @@ endif # BT_MESH_RELAY
 
 endif
 
-config BT_MESH_NLC_PERF_CONF
-	bool "Enable NLC performance configurations"
+config BT_MESH_NLC_PERF_BASELINE
+	bool
 	select BT_MESH_COMP_PAGE_2
 	select BT_MESH_COMP_PAGE_1
 	help
-	  Enables performance configurations required for NLC profiles.
-	  When implementing the NLC Basic Lightness Controller profile
-	  individually or in conjunction with other NLC profiles, also
-	  enable CONFIG_BT_MESH_NLC_PERF_LIGHTNESS_CTRL.
-
-choice BT_MESH_NLC_PERF_OPTIONS
-	prompt "NLC configuration options"
-	depends on BT_MESH_NLC_PERF_CONF
+	  Provides baseline performance requirements for NLC profiles.
+	  Automatically enabled when any NLC profile is selected.
 
-config BT_MESH_NLC_PERF_DEFAULT
-	bool "Default"
+config BT_MESH_NLC_PERF_STRICT
+	bool
+	select BT_MESH_NLC_PERF_BASELINE
 	help
-	  Enables performance configurations required for the NLC Basic Scene Selector,
-	  Dimming Control, Energy Monitor, Occupancy Sensor and Ambient Light Sensor profiles.
-	  When implementing the NLC Basic Lightness Controller profile individually or in
-	  conjunction with other NLC profiles, use CONFIG_BT_MESH_NLC_PERF_LIGHTNESS_CTRL instead.
+	  Provides strict performance requirements needed for some NLC profiles.
+	  Automatically enabled when relevant NLC profiles are selected.
+	  Includes all baseline requirements plus additional strict requirements.
+
+menu "NLC Profile performance configuration"
+
+config BT_MESH_NLC_PERF_SCENE_SELECTOR
+	bool "Basic Scene Selector NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_DIMMING_CONTROL
+	bool "Dimming Control NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_ENERGY_MONITOR
+	bool "Energy Monitor NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_OCCUPANCY_SENSOR
+	bool "Occupancy Sensor NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
+
+config BT_MESH_NLC_PERF_AMBIENT_LIGHT_SENSOR
+	bool "Ambient Light Sensor NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_BASELINE
 
 config BT_MESH_NLC_PERF_LIGHTNESS_CTRL
-	bool "Enable strict performance configurations required for the Basic Lightness Controller"
+	bool "Basic Lightness Controller NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_STRICT
+
+config BT_MESH_NLC_PERF_HVAC_INTEGRATION
+	bool "HVAC Integration NLC profile performance configuration"
+	select BT_MESH_NLC_PERF_STRICT
+
+endmenu
```

Comment:
Nitpick, and I'm not 100% sure what I think is best here, but have you considered moving the old, deprecated options into this menu as well?

Reply:
I do not think they should be part of menu since this gives wrong assumption this is one of the options. However, this is just the same but with old style.